cas-server-support-oauth/src/main/java/org/jasig/cas/config/OAuthConfiguration.java

```diff
@@ -51,14 +73,35 @@
     public Config oauthSecConfig() {
         final CasClient oauthCasClient = new CasClient(this.casLoginUrl);
         oauthCasClient.setName(CAS_OAUTH_CLIENT);
+        oauthCasClient.setCallbackUrlResolver((url, context) -> {
```

Comment:
The callback resolver here attempts to append the entityID and the redirect_uri parameters to to the callback. This allows the CAS validation process to then:

1. Detect the fact that the callback url is really not the real service, but it is an OIDC request with the real target that is the redirect_uri. It knows this is an OIDC request because it can detect the callback url based on those parameters as an OIDC thing. 
2. We pick the redirect_uri used during the validation process to use that definition to process user attributes and claims. The redirect_uri is what is registered in CAS service defns. This is consistent with how SAML works where entityID are registered, and the callback url there is also modified to remember and note the entityID to whom attributes shall be released. 

Right now, it is my understand that attributes for OAuth are processed based on the definition of the callback url, which means, they can never be released...or it may be the case that everything is released since we don't do proper filtering based on the redirect_uri and its attribute release policy.